pkg/config/collector/collector.go

```diff
@@ -63,9 +63,141 @@ func (c *Config) MergeConfigURL() error {
 	return c.MergeConfig(remoteConfig)
 }
 
+func (c *Config) toMap() (map[string]interface{}, error) {
+	var result map[string]interface{}
```

Comment:
`Config` is a `map[string]interface{}`. Can't we use it as it is? (maybe with type casting or directly)

Reply:
Yeah I guess it should be possible but I didn't go with that for two reasons:

1. For iterations of the items, I still had to convert to `map[string]interface{}`
2. It feels weird to put everything on a `Config` for example, when I'm deep merging the map inside an array, that is not a full config but just a map in an array, so to keep `DeepMerge` as generic as possible I decided to only use maps and not having the need to know anything about `Config`